lib/lint.js

```diff
@@ -0,0 +1,70 @@
+const { validateHubl } = require('@hubspot/cli-lib/api/validate');
+const { getPortalId } = require('@hubspot/cli-lib');
+const vscode = require('vscode');
+
+const getRange = (document, error) => {
+  if (error.startPosition > 0) {
+    return document.getWordRangeAtPosition(
+      new vscode.Position(error.lineno - 1, error.startPosition - 1)
+    );
+  } else {
+    return document.lineAt(new vscode.Position(error.lineno - 1, 0)).range;
+  }
+};
+
+async function updateDiagnostics(document, collection) {
+  const { renderingErrors } = await validateHubl(
+    getPortalId(),
+    document.getText()
+  );
+
+  const fitleredErrors = renderingErrors.filter((error) => {
+    return error.reason != 'MISSING';
```

Comment:
Do we have consts defined for the error reasons? Or is that defined in the BE?

Reply:
Defined on BE, but since you pointed it out I added the possible values to `constants.js`. I'm mainly concerned with focusing out `missing` files for now and although VS Code provides different severity levels, the HS `templateError` types all seem like syntax `Error`s